File: 199-binary-tree-right-side-view/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def rightSideView(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: List[int]
        """
        if not root:
            return []

        res =[]
        q = deque([root])

        while q:
            right =None

            for i in range(len(q)):
                node = q.popleft()

                right = node #Right most of this level
            
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)

            res.append(right.val)
        return res

        

   

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        if (root == null){
            return new ArrayList<Integer>();
        }

        ArrayDeque<TreeNode> nextLevel = new ArrayDeque(){
            {
                offer(root);
            }
        };
        ArrayDeque<TreeNode> curr = new ArrayDeque();
        List<Integer> ans = new ArrayList();

        TreeNode node = null;
        while(!nextLevel.isEmpty()){
            curr = nextLevel;
            nextLevel = new ArrayDeque<>();

            while(!curr.isEmpty()){
                node = curr.poll();
                if(node.left != null){
                    nextLevel.offer(node.left);
                }
                if(node.right != null){
                    nextLevel.offer(node.right);
                }
            }
            ans.add(node.val);
        }
        return ans;
        
    }
}